exports/exports_test.go

```diff
@@ -119,6 +119,12 @@ var _ = Describe("The public API", func() {
 		Entry("by improper created at", "created_at=spring", "", http.StatusBadRequest),
 		Entry("by expires", "expires_at=2023-01-01T00:00:00Z", "", http.StatusOK),
 		Entry("by improper expires", "expires_at=nextyear", "", http.StatusBadRequest),
+		Entry("by application", "application=exampleApp", "Test Export Request 1", http.StatusOK),
+		Entry("by resource", "resource=exampleResource", "Test Export Request 1", http.StatusOK),
+		Entry("by application that doesn't exist", "application=notAnApp", "", http.StatusOK), // containing an empty string is not the best way
+		Entry("by resource that doesn't exist", "resource=notAResource", "", http.StatusOK),
+		Entry("by application and resource", "application=exampleApp&resource=exampleResource", "Test Export Request 1", http.StatusOK),
```

Comment:
I think we might need to build out the test a little more for this, either here or in a second test case so that we can make sure the filtering includes and excludes the right exports. Here we're only checking that `Test Export Request 1` is included in the result, but that doesn't tell us that the filter is working as expected.

I'm thinking something like this, but let me know what you think:

```
func populateTestData() chi.Router {
	// define router
	router := setupTest(mockReqeustApplicationResouces)

	for i := 1; i <= 5; i++ {
		req := createExportRequest(
			fmt.Sprintf("Test Export Request %d", i),
			"json",
			"",
			`{"application":"exampleApp", "resource":"exampleResource"}`,
		)

		rr := httptest.NewRecorder()

		AddDebugUserIdentity(req)
		router.ServeHTTP(rr, req)
	}

	oneDayAgo := time.Now().AddDate(0, 0, -1)
	oneDayFromNow := time.Now().AddDate(0, 0, 1)

	modifyExportCreated("Test Export Request 1", oneDayAgo)
	modifyExportCreated("Test Export Request 2", oneDayFromNow)
	modifyExportExpires("Test Export Request 3", oneDayFromNow)

	// Used for testing filtering by application and resource
	modifyExportSources("Test Export Request 4", "exampleApp", "exampleResource")
	modifyExportSources("Test Export Request 5", "exampleApp2", "exampleResource2")

	return router
}

// modify the application and resource of all sources in a given export
func modifyExportSources(exportName string, application string, resource string) {
	var exportPayload models.ExportPayload
	testGormDB.Where("name = ?", exportName).First(&exportPayload)

	var sources []models.Source
	testGormDB.Find(&sources)

	for _, source := range sources {
		if source.ExportPayloadID == exportPayload.ID {
			source.Application = application
			source.Resource = resource
			testGormDB.Save(&source)
		}
	}
}
```

Reply:
Thanks! yeah it's best to make it more specific. I added your modifyExportSources function to the test and made the count and multiple export checks.